code/modules/shuttle/shuttles/escape_shuttle.dm

```diff
@@ -71,11 +75,78 @@
 		door_handler.control_doors("force-unlock")
 		return
 
+	if(prob(CRASH_LAND_PROBABILITY))
+		create_crash_point()
+	else
+		destination = null
+
 	set_mode(SHUTTLE_IGNITING)
 	on_ignition()
 	setTimer(ignitionTime)
 	launched = TRUE
 
+/obj/docking_port/mobile/escape_shuttle/proc/create_crash_point()
+
+	var/ground_z_level
+
+	var/z_level_index = 1
+	for(var/datum/space_level/level in SSmapping.z_list)
+		if(ZTRAIT_GROUND in level.traits)
+			ground_z_level = z_level_index
+			break
+		z_level_index++
+
+	var/list/potential_areas = SSmapping.areas_in_z[SSmapping.areas_in_z[ground_z_level]]
+
+	var/area/area_picked = potential_areas[rand(1, potential_areas.len)]
+
+	var/list/potential_turfs = list()
+
+	for(var/turf/turf_in_area in area_picked.contents)
+		potential_turfs += turf_in_area
+
+	if(!potential_turfs.len)
+		destination = null
+		return
+
+	var/turf/turf_picked = potential_turfs[rand(1, potential_turfs.len)]
+
```

Comment:
Need extra handling for edge cases, here are some things that come to mind:
 * checking area `can_build_special` to avoid landing in Out-of-Bounds areas
 * checking turf density to avoid landing in completely closed areas
 * checking the picked turf won't project the pod landing beyond map border / maximum size